tests/dialects/test_stencil.py

```diff
@@ -560,3 +570,74 @@ def test_stencil_access():
     assert isinstance(access, AccessOp)
     assert access.offset == offset_index_attr
     assert access.temp.typ == temp_type
+
+
+# TODO: Move to a notebook at some point with proper documentation
+def test_1d3pt_stencil_construct():
+    """
+    An example 1d-3pt stencil implementation from XDSL Python-land
+    using the stencil dialect
+    """
+
+    shape = (8,)
+    space_order = 2
+    r = space_order // 2
+    symbolic_shape = (0 - r, shape[0] - r)
+
+    # Domain with halo shape
+    temp0 = TempType(len(shape), f32)
+
+    # Computational domain shape
+    field0 = FieldType([symbolic_shape], f32)
+
+    # The computation block
+    block0 = Block(arg_types=[temp0])
+
+    # Stencil computation
+    stencil_ops = [
+        stencil_acs_l := AccessOp.get(block0.args[0], (-1,)),
+        stencil_acs_c := AccessOp.get(block0.args[0], (0,)),
+        stencil_acs_r := AccessOp.get(block0.args[0], (1,)),
+        stencil_comp0 := Addf(stencil_acs_l, stencil_acs_c),
+        stencil_comp1 := Addf(stencil_comp0, stencil_acs_r),
+        # Define the return operation
+        return_op_ := ReturnOp.get([stencil_comp1]),  # type: ignore
```

Comment:
Why the type ignore?

Reply:
Because otherwise pyright complains about the unused variable, but declaring one felt more consistent and readable

Reply:
But it should probably go away if this is migrated to the builder, which I would also recommend :)

Reply:
Yeah I would use the builder for this either way too; are the new with blocks merged already?
(Otherwise I like the underscore suggestion, thanks!)

Reply:
The with blocks are merged, and are probably the thing to use. I'm still trying to work out ergonomics for optional regions

Reply:
@superlopuh could you give me a pointer on that?

Reply:
The best examples are probably in the pytest pdl interpreter tests, `test_pdl_interpreter.py`

Reply:
Oh no, just rewrote it with the implicit stuff, not finding the with blocks :smiling_face_with_tear: Will have a look

Reply:
Okay, it was easy actually, done